File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
    var answer = 0;
    const clothes_dict = {};
    clothes.forEach(([value, key]) => {
        clothes_dict[key] = (clothes_dict[key] || 0) + 1;
    });
    const values = Object.values(clothes_dict)
    answer = values.reduce((acc, cur) => {
        console.log(acc)
        return acc*(cur+1)
    }, 1);
    return answer - 1;
}

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    static int answer = 1;
    public int solution(String[][] clothes) {
        HashMap<String, Integer> counts = new HashMap<>();
        
        for (String[] clothe: clothes) {
            String name = clothe[0];
            String clotheType = clothe[1];
            int cnt = counts.getOrDefault(clotheType, 0);
            counts.put(clotheType, cnt + 1);
        }
        counts.forEach((k, v) -> {
            answer *= (v+1);
        });
        return --answer;
    }
}